core/src/main/java/tech/tablesaw/io/FileReader.java

```diff
@@ -168,20 +180,44 @@ private void addRows(
                 + " expected.");
       }
 
-      // append each column that we're including (not skipping)
-      int cellIndex = 0;
-      for (int columnIndex : columnIndexes) {
-        Column<?> column = table.column(cellIndex);
-        AbstractColumnParser<?> parser = parserMap.get(column.name());
-        try {
-          String value = nextLine[columnIndex];
+      int samplesCount = table.rowCount();
+      if (sampleSize == -1 || samplesCount < sampleSize) {
```

Comment:
I might do `sampleSize < 0` rather than `sampleSize == -1` to be a bit more tolerant of input values

Reply:
Fixed.